pkg/shares/split_compact_shares.go

```diff
@@ -78,59 +67,42 @@ func (css *CompactShareSplitter) WriteTx(tx coretypes.Tx) {
 func (css *CompactShareSplitter) write(rawData []byte) {
 	if css.done {
 		// remove the last element
-		if !css.isEmptyPendingShare() {
+		if !css.shareBuilder.IsEmptyShare() {
 			css.shares = css.shares[:len(css.shares)-1]
 		}
 		css.done = false
 	}
 
-	css.maybeWriteReservedBytesToPendingShare()
+	if err := css.shareBuilder.MaybeWriteReservedBytes(); err != nil {
+		panic(err)
+	}
```

Comment:
[optional]

leaving as optional since we should only ever reach this point if there's an invalid state, but it does limit the safe uses of this library. 

I think there are a few places where we could bubble up these errors with the only consequence being something that's slightly less aesthetic, since all of there uses are in functions that can or already are returning an error

Reply:
good idea. 
I am going to get rid of all panics in a separate PR.